src/components/constraints/Constraints.svelte

```diff
@@ -133,6 +113,93 @@
     }
   }
   $: constraintModelId = getConstraintModelId(selectedConstraint);
+  $: constraintsEditPermissionsMap = ($constraintsAll ?? []).reduce(
+    (prevMap: ConstraintsEditPermissionsMap, constraint: Constraint) => {
+      const { model_id, plan_id } = constraint;
+
+      if (plan_id !== null) {
+        const plan = initialPlanMap[plan_id];
+        if (plan) {
+          return {
+            ...prevMap,
+            plans: {
+              ...prevMap.plans,
+              [plan_id]: featurePermissions.constraints.canUpdate(user, plan, constraint),
+            },
+          };
+        }
+      } else if (model_id !== null) {
+        const model = initialModelMap[model_id];
+        if (model) {
+          return {
+            ...prevMap,
+            models: {
+              ...prevMap.models,
+              [model_id]: model.plans.reduce((prevPermission: boolean, { id }) => {
+                const plan = initialPlanMap[id];
+                if (plan) {
+                  return prevPermission || featurePermissions.constraints.canUpdate(user, plan, constraint);
+                }
+                return prevPermission;
+              }, false),
+            },
+          };
+        }
+      }
+
+      return prevMap;
+    },
+    {
+      models: {},
+      plans: {},
+    },
+  );
+  $: {
+    hasPermission = initialPlans.reduce((prevPermission: boolean, plan) => {
+      return prevPermission || hasPlanPermission(plan, user);
+    }, false);
+    columnDefs = [
+      ...baseColumnDefs,
+      {
+        cellClass: 'action-cell-container',
+        cellRenderer: (params: ConstraintsCellRendererParams) => {
+          const actionsDiv = document.createElement('div');
+          actionsDiv.className = 'actions-cell';
+          new DataGridActions({
+            props: {
+              deleteCallback: params.deleteConstraint,
+              deleteTooltip: {
+                content: 'Delete Constraint',
+                placement: 'bottom',
+              },
+              editCallback: params.editConstraint,
+              editTooltip: {
+                content: 'Edit Constraint',
+                placement: 'bottom',
+              },
+              hasDeletePermission: params.data ? hasEditPermission(user, params.data) : false,
```

Comment:
Should this be a "hasDeletePermission" call? The trashcan is still visible, but deleting is not allowed.

Reply:
Effectively the delete and edit permissions are the exact same, which is why I just used one function. Did you get an error from the backend saying that deleting wasn't allowed?

Reply:
I see what you're seeing. I'll split it out just in case, but there might be another inconsistency with permissions. I'll look into it. Thanks!

Reply:
Okay did some more investigation. It's happening if as a user role, I create a plan constraint with no owner (thus admin), but my user owns the plan the constraint is set to.

Reply:
@camargo it looks like it's a backend permissions issue. @Mythicaeda will do a fix